### PR TITLE
Manage jars with lock jar

### PR DESCRIPTION
**Pull request by mguymon**
Uses [Lock Jar](https://github.com/mguymon/lock_jar) to handle the jar dependencies, removing the requirement if having to package the uber jar with the gem. RubyGem will rejoice not having to store a ~ 5meg jar with every release.

The other perk is, down stream projects can do dependency resolution with Telekinesis via Lock Jar. This was actually a problem I had were another jar dependency was conflicting with the guava version. 

The jar dependencies are installed to the local maven dir the first time [LockJar.load(resolve: true)](https://github.com/mguymon/telekinesis/blob/cf6019eae11772c5c0fabc83fb679bacb086094c/lib/telekinesis.rb#L7-L8). After that, the local jars are used. It is possible to pre fetch the jars when the gem is installed, but it will require [a Rakefile as the gem's extension](https://github.com/mguymon/telekinesis/blob/cf6019eae11772c5c0fabc83fb679bacb086094c/lib/telekinesis.rb#L7-L8)

For this to work, the Telekinesis jar will have to be [deployed to maven central](http://central.sonatype.org/pages/ossrh-guide.html) which is a bit of a PITA and requires a pom with a bunch of extra details. I went ahead it updated the pom xml and added a profile that sign's the jar when it deploys (otherwise it wants to sign it every time you package it). Releasing to maven central works by `mvn deploy -P release-sign-artifacts`.

To test locally, you have to do a `mvn install` first (until the jar is released to maven central). After that, all the specs pass as the jars are automatically loaded based on the `Jarfile.lock`.

### CODE REVIEW

File: ext/pom.xml [hunk 1]
```diff
@@ -2,10 +2,35 @@
...
+    <url>https://github.com/kickstarter/telekinesis</url>
     <groupId>com.kickstarter</groupId>
     <artifactId>telekinesis</artifactId>
     <version>3.1.0</version>
 
+    <licenses>
+       <license>
+           <name>TBD</name>
+           <url>TBD</url>
+           <distribution>repo</distribution>
```
Comment (mguymon) [inline ext/pom.xml]: Need to define what license your code falls under

**Comment by mguymon** [comment]
CI failed with

> Failed to collect dependencies for [com.kickstarter:telekinesis:jar:3.1.0 (compile)]

Because the jar could not be found locally or remotely.

File: ext/pom.xml (hunk 1 above)
Comment (blinsay) [inline ext/pom.xml]: Thanks for pointing this out! Looks like we forgot our standard license stanza in the README. I just [added it](https://github.com/kickstarter/telekinesis#license).

**Comment by blinsay** [comment]
> Uses Lock Jar to handle the jar dependencies, removing the requirement if having to package the uber jar with the gem. RubyGem will rejoice not having to store a ~ 5meg jar with every release.

> The other perk is, down stream projects can do dependency resolution with Telekinesis via Lock Jar. This was actually a problem I had were another jar dependency was conflicting with the guava version.

Thanks for being concerned about our dependency hygiene and the rubygems infrastructure. I'm gonna think on this over the weekend.

In the meantime I'm going to clean up the Guava dependency - it's not getting used by the extension at all.

**Comment by blinsay** [comment]
After talking about this internally, we've decided to hold off on switching to Lock Jar for now. Sticking to the bundler-only happy path seems a little more friendly. If we keep running into folks who have issues with Java dependency management, we'll definitely reconsider.

Thanks for bringing this up!

**Pull request closed without merging**